compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/BinaryFileWriter.java

```diff
@@ -80,19 +94,23 @@ public ProgramFile genExecutable(BLangPackage entryPackageNode) {
     }
 
     public void writeExecutableBinary(BLangPackage packageNode) {
-        String fileName = getOutputFileName(packageNode, BLANG_EXEC_FILE_SUFFIX);
-        writeExecutableBinary(packageNode, fileName);
-
-        // Generate balo
         Path path = this.sourceDirectory.getPath();
-        if (Files.isDirectory(path.resolve(ProjectDirConstants.DOT_BALLERINA_DIR_NAME))) {
-            ProjectSourceRepo projectSourceRepo = new ProjectSourceRepo(path);
-            Patten packageIDPattern = projectSourceRepo.calculate(packageNode.packageID);
-            Stream<Path> pathStream = packageIDPattern.convert(projectSourceRepo.getConverterInstance());
-            pathStream = Stream.concat(pathStream, packageIDPattern.sibling(path("Ballerina.md")).convert
-                    (projectSourceRepo.getConverterInstance()));
-            String prjPath = projectSourceRepo.getConverterInstance().toString();
-            ZipUtils.generateBalo(packageNode, prjPath, pathStream);
+        ProjectSourceRepo projectSourceRepo = new ProjectSourceRepo(path);
+
+        String prjPath = projectSourceRepo.getConverterInstance().toString();
+        if (!this.listPackageDependencies && !this.noWriteDisc) {
```

Comment:
I think we need to move this check up the call stack, ie org.wso2.ballerinalang.compiler.Compiler#build()

Reply:
Fixed with 5985a73